ee/consoleuser/consoleuser_linux.go

```diff
@@ -73,3 +63,38 @@ func CurrentUids(ctx context.Context) ([]string, error) {
 
 	return uids, nil
 }
+
+// listSessions execs `loginctl list-sessions` in order to retrieve the current list of sessions.
+// Depending on the systemd version, we have to use different flags to output the results as JSON.
+// We may want to attempt parsing the output regardless in the future -- see launcher #1522.
+func listSessions(ctx context.Context) (listSessionsResult, error) {
+	var sessions listSessionsResult
+
+	// Try with `--output=json` first, to support the more widely-used older versions of systemd
+	legacyCmd, err := allowedcmd.Loginctl(ctx, "list-sessions", "--no-legend", "--no-pager", "--output=json")
+	if err != nil {
+		return nil, fmt.Errorf("creating loginctl command --no-legend --no-pager --output=json: %w", err)
+	}
+	legacyOut, err := legacyCmd.Output()
+	if err == nil {
```

Comment:
Do we need this `if`?

Reply:
No point in trying to unmarshal if the command failed, right -- or have I missed something?

Reply:
Of the command failed, it returns on line 76

Reply:
That's the creation of the exec.Cmd struct -- we return on 76 if we fail to validate the command (e.g. if we can't find the `loginctl` binary). But we don't exec the command till line 78, so we're checking for exec failures here.

Reply:
Oh, I see it now. This is inverted logic -- no error early returns. You could probably combine, `if legacyOut, err := legacyCmd.Output();  err == nil {` but I don' think that changes anything real. 

Thank you for putting up with my fuzziness.